### PR TITLE
Fix broken manifest workflow

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Fix broken manifest workflow: https://build.ci.opensearch.org/job/manifest-update/49/console
and modify the code to update the `osd-increment-plugin-versions.yml` workflow with new branch entry.

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/3610

### Test on local
Updated the workflow files
```

Adding 2.9 to /usr/share/opensearch/git/opensearch-build/.github/workflows/os-increment-plugin-versions.yml
2023-06-08 21:48:16 INFO     Added new version to the version increment workflow
Adding 2.9 to /usr/share/opensearch/git/opensearch-build/.github/workflows/osd-increment-plugin-versions.yml
2023-06-08 21:48:16 INFO     Added new version to the version increment workflow
```

git status: Added new manifests
```

# On branch main
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#	modified:   .github/workflows/os-increment-plugin-versions.yml
#	modified:   .github/workflows/osd-increment-plugin-versions.yml
#	modified:   jenkins/check-for-build.jenkinsfile
#
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#	manifests/2.8.1/
#	manifests/2.9.0/

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/tests_manifests_workflow/test_input_manifests.py [hunk 1]
```diff
@@ -129,10 +129,16 @@ def test_add_to_cron(self, mock_open: MagicMock) -> None:
             f"TARGET_JOB_NAME=distribution-build-test;BUILD_PLATFORM=linux;BUILD_DISTRIBUTION=tar\n"
         )
 
-    def test_versionincrement_workflow(self) -> None:
+    def test_os_versionincrement_workflow(self) -> None:
         self.assertEqual(
-            InputManifests.versionincrement_workflow(),
-            os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", ".github", "workflows", "increment-plugin-versions.yml"))
```
Comment (gaiksaya) [inline tests/tests_manifests_workflow/test_input_manifests.py:135]: Wondering why didn't the test fail when we renamed the workflow. Any idea?

Comment (prudhvigodithi) [reply tests/tests_manifests_workflow/test_input_manifests.py:135]: The test ensures the the name of the file do match from the defined @classmethod, does not test if the file exists on the file system.
Example the test `test_os_versionincrement_workflow` asserts the file name same as defined in  @classmethod `os_versionincrement_workflow`.

Comment (gaiksaya) [reply tests/tests_manifests_workflow/test_input_manifests.py:135]: I see! Is there anyway we can add tests to make sure the file exists on the system?

**Comment by rishabh6788** [comment]
Please replace `print` with `logging` if required.

File: src/manifests_workflow/input_manifests.py [hunk 2]
```diff
@@ -192,29 +196,31 @@ def add_to_cron(self, version: str) -> None:
         logging.info(f"Wrote {jenkinsfile}")
 
     def add_to_versionincrement_workflow(self, version: str) -> None:
```
Comment (rishabh6788) [inline src/manifests_workflow/input_manifests.py:198]: Please add test for `add_to_versionincrement_workflow` method as well.

Comment (prudhvigodithi) [reply src/manifests_workflow/input_manifests.py:198]: Added Rishab

**Review by gaiksaya — APPROVED** [review]

File: tests/tests_manifests_workflow/test_input_manifests.py (hunk 1 above)
Comment (prudhvigodithi) [reply tests/tests_manifests_workflow/test_input_manifests.py:135]: Added Sayali, thanks

**Pull request merged**